### PR TITLE
Error Messages: Refactor backticks, capitalize messages

### PR DESCRIPTION
**Pull request by sleepiejohn**
This PR removes unnecessary backticks ` from messages and explanations, around interpolations and from text, wraping those texts into interpolations so it can be picked by the highlighter

- `` `()`  `` becomes `${"()"}`
- `` `$mySymbol` `` becomes `$mySymbol`

Also capitalizes messages, but:
- Messages that starts with keywords such as `match` were not capitalized.
- Messages starting with symbol name such as `value`, `class`, `object` were not capitalized.

This all to remove clutter and increase experience when sending those errors to editors for example, and do let newcomers be confused since `` ` `` is part of the language when used in a case expression and to escape keywords 

Part of #1589

### CODE REVIEW

**Comment by sleepiejohn** [comment]
Waiting on tests to perform on CI so I can fix any broken, since on my machine I can't compile the master branch `master`

**Comment by sleepiejohn** [comment]
Tests fixed!

**Comment by nicolasstucki** [comment]
@sleepiejohn could you rebase your PR the changes so far look good.

File: compiler/test-resources/repl/errmsgs [hunk 1]
```diff
@@ -3,68 +3,68 @@ scala> class Inv[T](x: T)
...
-  |                              required: String
+  |                              Found:    Inv[Int]
+  |                              Required: String
 scala> val b: Inv[String] = new Inv(1)
 1 | val b: Inv[String] = new Inv(1)
   |                              ^
-  |                              found:    Int(1)
-  |                              required: String
+  |                              Found:    Int(1)
+  |                              Required: Stri
```
Comment (nicolasstucki) [inline compiler/test-resources/repl/errmsgs]: You removed part of `String` by mistake here.

**Comment by sleepiejohn** [comment]
Seems that language server tests are now included and it's currently broke because of capitalization, I'll tend to it later today

**Comment by nicolasstucki** [comment]
Thanks @sleepiejohn

**Pull request merged**